### DISCUSSION TITLE
URL rewrites for subdomains

### DISCUSSION THREAD

**Discussion by cosbgn** [vercel/vercel#6443] [Help]
Hello,

I have an app which has the following routes: `https://example.com/app/random-app-id` now I want to use a separate service to allow my customers to connect their domains to their app. To do that I need the app-id to be mapped as a subdomain. I noticed that Vercel supports * subdomains. So I was wondering can I write a rewrite which maps: `app-id.example.com` to `https://example.com/app/app-id`? If yes, how? Thanks!

**Comment by josezy** [comment]
Did you try this guide?
https://vercel.com/support/articles/can-i-redirect-from-a-subdomain-to-a-subpath

**Comment by skasyn** [comment]
If you need to handle dynamic subdomain, you should take a look to this [github comment](https://github.com/vercel/next.js/discussions/12848#discussioncomment-680225)
